### ISSUE TITLE
Feature Request: Long Text Plot

### ISSUE THREAD

**Issue by darul75** [piccolomo/plotext#127]
Hello,

I would like to display some text, raw text in a block, like a list of items.

In the following screenshot you can see `bel` instead of `label`

<img width="1380" alt="Screenshot 2022-09-04 at 21 12 18" src="https://user-images.githubusercontent.com/2134003/188329824-01813ff3-bf65-4508-bcd0-2798bd084d03.png">

A final result being more likely like:

`label1........on first line`
`label2.....on secondline`
`...`

Being able to pass a string with carriage return separators would be ideal or that the text wraps automatically.

Let me know if that is not clear enough and if it sounds feasible.

Been using the following code

`plt.text('label', 1, 1, xside= 'lower', yside = 'left', color = 'red'); \`

**Comment by piccolomo** [comment]
i reply quickly and i hope i understood the issue correctly. try to use the
parameter alignment in the text() function, particularly passing "left".
did it solve it?

**Comment by piccolomo** [comment]
the issue should be due to default center alignment

**Comment by piccolomo** [comment]
btw you do not generally need xside and yside parameters as they accept the default values you passed

**Comment by piccolomo** [comment]
btw the automatic wrapping idea, is something i could definitely think about. thanks for suggesting

**Comment by piccolomo** [comment]
Perhaps a text box function could be more useful for large text. Because text() is more for labelling plots where needed, but for dedicated text probably better a text box utility... not trivial but definitively useful. Any coding actual idea is welcome, but for now i will go back mostly to lethargy mode for awhile, after latest update

**Comment by piccolomo** [comment]
Thinking out loud: would it be enough just plain text in the box? Or better with color and style, and alignment? In which care it will be more complicated. 

One option is to integrate the package "rich" to deal with beautiful texts, but it will require a dependency, the other require some coding, non trivial one. In second case, how would the user indicate the coloring/style/alignment of a particular part of the string? I was thinking of something like "[red] some red text [/red] " which I think is similar to what rich does. Any input welcomed.

**Comment by piccolomo** [comment]
Hi @darul75,

thanks for issue report, I reply to you now properly. Here are some points:

- try to use the parameter alignment in the `text()` function, particularly passing `"left"`: did it work?
- generally you do not need `xside `and `yside `parameters as they accept the default values you passed,
- perhaps a text box function could be a good idea as the function `text()` is more for labelling plots where needed. The complication is about coloring, style and alignment inside the text box. 

Thanks,
Savino

**Comment by darul75** [comment]
Hi @piccolomo 

Been looking at the codebase if I could raise a PR but my knowledge of Python or how this thing works is difficult for me :)

I have tried the following

`plt.text('label1 ', 1, 1, yside = 'left', color = 'red'); \`

gives me a correct display but a longer text seems to be truncated.

`plt.text('label1 is a very long text', 1, 1, yside = 'left', color = 'red'); \`

![Screenshot 2022-09-09 at 10 15 44](https://user-images.githubusercontent.com/2134003/189304675-46b1f024-6198-4e04-b20d-e01cff92d662.png)

**Comment by piccolomo** [comment]
Hi @darul75 ,

try `plt.text('label1 ', 1, 1, alignment = 'left'); ` instead. Does it help?

**Comment by darul75** [comment]
may have passed the wrong arguments indeed, will check.

**Comment by piccolomo** [comment]
Hi @darul75,

I have updated GitHub version to solve your other issue of having multiple lines in the same text box. I manage to do it: all you could do now is have a single string with `'\n'` separator between each sub string, like in this example:
![image](https://user-images.githubusercontent.com/56689657/189652239-2935056c-2ed1-46dd-a1bd-eb7eddea3b28.png)

The new version is available [on GitHib](https://github.com/piccolomo/plotext) for now and soon also [on PyPi](https://pypi.org/project/plotext/). To install follow the indications [here](https://github.com/piccolomo/plotext#install).

Any feedback is welcomed. Please report if the issue persists. 

Thanks a lot for your inputs and all the best,
Savino

**Comment by darul75** [comment]
just awesome again...gonna really help.me finishing that dashboard

**Comment by piccolomo** [comment]
Ok glad it helps, I close this for now. PyPi version will be updated in following weeks. Thanks for feedback, Savino.

**Comment by darul75** [comment]
Hello @piccolomo,

I tested your example with latest version of plotext.

Within a subplot layout I obtain the following where it goes to the previous plot on the left.

<img width="1030" alt="Screenshot 2022-09-17 at 21 37 15" src="https://user-images.githubusercontent.com/2134003/190873819-6b6d2a93-b042-449f-b8bf-210de5b5908b.png">

I really love your lib by the way, very handy.

**Comment by piccolomo** [comment]
I reopen so i remember to dealt with it later

**Comment by piccolomo** [comment]
I reopen so i remember to deal with it later

**Comment by piccolomo** [comment]
Hi @darul75 , this issue should be solved in new GitHub version. 

To install please follow the indications [here](https://github.com/piccolomo/plotext#install).

Thanks a lot for your inputs and all the best,
Savino

**Comment by darul75** [comment]
Perfect timing I am just about to finish what I was building, will share it with you once it is done.

**Comment by piccolomo** [comment]
nice result!

**Comment by sparkwj** [comment]
![Untitled](https://user-images.githubusercontent.com/5135832/201260796-47fe99fc-99da-4fc7-9508-72b31b12b2eb.jpg)
hi @piccolomo，it seems the chinese text adjustment is not work, maybe wrong to calculate the characters length?

**Comment by piccolomo** [comment]
Hi @sparkwj  
I have opened a new issue about it [here](https://github.com/piccolomo/plotext/issues/158).  
Thanks
